### DISCUSSION TITLE
Upload object from database function

### DISCUSSION THREAD

**Discussion by diegonc** [supabase/supabase#28709] [Questions]
I'm not able to find a way to upload an object to supabase storage from a database function. The function has access to the following fields:

  - `object_id   text`
  - `object_data bytea`

and should be able to create an entry in the storage, under a fixed bucket, named `object_id` and with the data `object_data` associated to it.

Is it possible to do so? I need to call this function from a trigger and haven't found in the documentation how to interact with upload from plpgsql.

**Comment by GaryAustin1** [comment]
It is not possible directly.  You have to use the REST API with a path to get/store any object data.  You can use the http or pg_net extension to do this making a storage API REST call.

**Reply by diegonc** [reply]
Yeah, I was looking how to do it with [pgsql-http](https://github.com/pramsey/pgsql-http) but it doesn't seem to support multipart form data :-(

I'll check pg_net, didn't know about it.

**Reply by GaryAustin1** [reply]
You might also consider an edge function.  But you also might want to relook at how you are doing this and start with an edge function or handle the storage operation outside of the database if possible.

**Reply by diegonc** [reply]
Hmm, I don't think an edge function will help, as the transaction is not committed yet if the function gets called from the trigger. Unless I'm able to delay execution until the transaction is committed.
